corehq/form_processor/tests/utils.py

```diff
@@ -280,19 +293,25 @@ def create_form_for_test(
     """
     from corehq.form_processor.utils import get_simple_form_xml
 
-    form_id = uuid4().hex
+    form_id = form_id or uuid4().hex
     utcnow = received_on or datetime.utcnow()
+    kwargs.setdefault('xmlns', 'http://openrosa.org/formdesigner/form-processor')
 
-    form_xml = get_simple_form_xml(form_id, case_id)
+    if form_data is not None:
+        form_xml = convert_form_to_xml(form_data)
+        if user_id is None and form_data.get('meta'):
+            user_id = form_data['meta'].get('userID', user_id)
+    else:
+        form_xml = get_simple_form_xml(form_id, case_id)
 
     form = XFormInstanceSQL(
         form_id=form_id,
-        xmlns='http://openrosa.org/formdesigner/form-processor',
         received_on=utcnow,
-        user_id=user_id,
+        user_id=user_id or 'user1',
```

Comment:
Any reason to add this here, rather than as a default argument? Similarly, why here and not for the `CommCareCaseSQL` below? Is a user even necessary here?